### DISCUSSION TITLE
Version 2 Fingerprint question. What does count=True do

### DISCUSSION THREAD

**Discussion by syedzayyan** [chemosim-lab/ProLIF#153] [General]
I don't necessarily understand what count does. To quote the docs 'but you can optionally enable the count parameter to enumerate all occurrences of an interaction', does it mean it generates all the possible interactions and displaying them can be controlled with `display_all=True`?

`fp_count = plf.Fingerprint(count=True)`

**Comment by cbouy** [comment]
Traditionally interaction fingerprints are bitvectors i.e. they only indicate the presence or absence of an interaction for a given ligand and residue pair. But for a given combination of ligand, protein residue and interaction type, there might be multiple combinations of atoms in the ligand and residue that satisfy the interaction constraints.

What `count=True` does is that it keeps track of all these combinations that satisfied the interaction constraints, so that it can indicate for example that there are 2 combinations of atoms in the ligand and residue XYZ that are involved in an HBond interaction where the ligand acts as a donor. 
Whereas `count=False` will  terminate early i.e. as soon as it finds atoms in the ligand-residue pair that satisfy an interaction, it sets the corresponding bit to `True/1` and stops here.

It's mostly useful for visualization purposes to be honest.

**Reply by syedzayyan** [reply]
Hello!

Thanks for the reply. I thought as much about the interactions. To be fair, it's very useful to know all the possible interactions in case of docking visualisations. Thank you very much for that. It has just come in time that I can tuck it into the thesis just in time.
